### ISSUE TITLE
Terrain Scanner not generating plants

### ISSUE THREAD

**Issue by novaktually** [Eladkay/Scanner#14]
Using terrain generator in Sky Factory 3.0.6 to generate chunks in Forest, Flower Forest, and Birch Forest biomes yields chunks with no trees, no grass, no flowers.

Edit: Forgot to mention this is with the ultimate biome scanner attached.

![2017-03-02_13 20 04](https://cloud.githubusercontent.com/assets/492648/23527634/a56eb986-ff4b-11e6-850a-eebc6f5a9637.png)

**Comment by Eladkay** [comment]
Seems like caves do generate.. so it's a problem in population. Try a different chunk?

**Comment by novaktually** [comment]
Still no luck. Pictured here: a Forest chunk, ForestHills chunks, and a Swampland chunk.

![2017-03-02_17 14 15](https://cloud.githubusercontent.com/assets/492648/23534298/d014700c-ff6b-11e6-8284-eb8ec36a9198.png)

**Comment by Eladkay** [comment]
I'll do my tests on my side. For now, if possible try to build a chunk close to the world spawn (in a creative world maybe)

**Comment by novaktually** [comment]
Two taiga chunks in creative world next to spawn:

![2017-03-03_07 27 27](https://cloud.githubusercontent.com/assets/492648/23556903/ff3b41d6-ffe2-11e6-9d95-fe2ddf5f1fa4.png)

**Comment by LatvianModder** [comment]
Missing biome decorator?

**Comment by Eladkay** [comment]
I can confirm that we are not decorating the biomes, however this isn't our responsibility. We are copying from another dimension which uses the vanilla generator. The way we could fix this is by loading the remote chunk like a player would.

**Comment by novaktually** [comment]
So, here's an interesting find. Don't know how helpful this is, but:

1. Scan chunk with terrain scanner: no plants.
2. Break scanner, replace in same spot, rescan: no plants.
3. Break scanner, replace in same chunk, different spot, rescan: plants.

![2017-03-03_14 07 20](https://cloud.githubusercontent.com/assets/492648/23570830/eed9b8ee-001a-11e7-943a-3af3c9d92db8.png)

**Comment by LatvianModder** [comment]
Maybe its related to this? https://www.reddit.com/r/feedthebeast/comments/5x0twz/investigating_extreme_worldgen_lag/
(Something about chunks only decorating when neighbor chunks have loaded)

**Comment by Eladkay** [comment]
You're right, this IS an interesting find and it tips me off on how to fix it.

**Comment by LatvianModder** [comment]
maybe you can force it to generate 4 overworld chunks, without placing blocks, to fool minecraft worlgen

**Comment by Eladkay** [comment]
That's what I was thinking, thing is, lag

**Comment by novaktually** [comment]
One additional thing I noticed (let me know if this should be opened as a new issue or not).

When rebuilding a chunk, new ores are being added to the existing chunk. If the chunk is being copied from a pre-generated chunk, I would have expected the ores to be in the same place.

For screengrab purposes, I shifted the second run of the terrain scanner one block to the left so you can more easily compare the before and after screens by following the progress of the build. Hope this is of some use!

![2017-03-05_07 35 25](https://cloud.githubusercontent.com/assets/492648/23588768/3680b9d8-0177-11e7-92cb-5728182db29f.png)
![2017-03-05_07 35 44](https://cloud.githubusercontent.com/assets/492648/23588769/36814448-0177-11e7-8298-51d7059731df.png)
![2017-03-05_07 36 41](https://cloud.githubusercontent.com/assets/492648/23588771/36820cd4-0177-11e7-8b85-b6e185cc4666.png)
![2017-03-05_07 36 55](https://cloud.githubusercontent.com/assets/492648/23588770/368204fa-0177-11e7-8978-c9b1116ec472.png)

**Comment by Eladkay** [comment]
Yeah, to prevent this exact issue and to make the scanner more viable, we generate some vanilla ores ourselves.

**Comment by DirtIsStupid** [comment]
Am having the same problem. When i select a chunk to scan using the map, or define a chunk using the queue, no plants generate. However, in an attempt to smooth out a diagonal line of chunks, i manually inputted coordinates offset by 8, and plants DID generate. Seems to work if more than one chunk is being referenced.
![2017-07-13_10 58 35](https://user-images.githubusercontent.com/30152758/28172893-de2b5c48-67ba-11e7-80a2-c2a64e3cbc28.png)

**Comment by Eladkay** [comment]
Could be that it just needs some time to load those chunks. It's weird but since it wasn't reported in 4 months and it's in a major pack, I'll assume it's not a huge issue.

**Comment by Zerger777** [comment]
Any new info on this? Having to redo all my SF3 generated chunks and I randomly get plants generating so I'm basically playing battleships over here. Except some chunks just blatantly refuse to generate leaving it akwardly naked. (Love the mod btw)
![2018-07-03_11 09 17](https://user-images.githubusercontent.com/40792330/42213751-902caaec-7eb1-11e8-9541-a938e7be3a7c.png)

**Comment by Eladkay** [comment]
See comment above you. I guess that if I tried to triangulate the issue it might nail down to one java version or GPU or CPU or windows version, but considering it’s uncommon and that I can’t reproduce it, you just got kind of unlucky. It’s a good chance to practice your landscaping skills!

**Comment by Zerger777** [comment]
Haha, damnit thats unfortunate. And oi my landskaping skillz are on the ball :P as you can see. Deco is a different story...
![2018-07-03_12 10 42](https://user-images.githubusercontent.com/40792330/42216659-846a701e-7eba-11e8-844b-1165eb962f5c.png)

**Comment by Zerger777** [comment]
Hi thought I'd put it down here that I did get it solved thanks to DirtIsStupid's comment. Mine did not work when overlapping only 2 chunks whilst generating but worked when overlapping 4 chunks at a 
 time. So using it smack in the middle of a chunk X:8 Z:8 for example then counting on 16 blocks every time. Doesn't work with the biome scanner attached, so no easy click a chunk to build there option; but using the queue add on or placing the terrain scanner on its own does. For those doing this too, get a calculator and a spreadsheet and be prepared for a lot of math. XD

**Comment by gdude2002** [comment]
Has there been any movement on this?

**Comment by IGCBOOM** [comment]
The 1.12.2 version hasn’t been released yet on the curse forge page but it is fixed in that version. I fixed it by offsetting the generation start by 8 and generating in the center of the chunk, to make it fit on the terrain scanner map I shifted the grid over as well. It is a crude fix but it generates vegetation with virtually no lag compared to normal.
